inst/include/csr_matrix_times_vector2.hpp

```diff
@@ -4,20 +4,14 @@
 /*
  * This works exactly like csr_matrix_times_vector but faster and less safe
  */
-template <typename T2__, typename T5__>
-inline
-Eigen::Matrix<typename boost::math::tools::promote_args<T2__, T5__>::type,
-              Eigen::Dynamic, 1>
-csr_matrix_times_vector2(const int& m,
-                         const int& n,
-                         const Eigen::Matrix<T2__, Eigen::Dynamic, 1>& w,
-                         const std::vector<int>& v,
-                         const std::vector<int>& u,
-                         const Eigen::Matrix<T5__, Eigen::Dynamic, 1>& b,
-                         std::ostream* pstream__) {
-  Eigen::Map<const Eigen::SparseMatrix<T2__,Eigen::RowMajor> >
-    sm(m, n, w.size(), &u[0], &v[0], &w[0]);
-  return sm * b;
+ template <typename T2__, typename T5__>
+ Eigen::Matrix<stan::promote_args_t<stan::value_type_t<T2__>,
+ stan::value_type_t<T5__>>, -1, 1>
+ csr_matrix_times_vector2(const int& m, const int& n, const T2__& w_arg__,
+                          const std::vector<int>& v,
+                          const std::vector<int>& u, const T5__& b_arg__,
+                          std::ostream* pstream__) {
+  return stan::math::csr_matrix_times_vector(m, n, w_arg__, v, u, b_arg__);
```

Comment:
@SteveBronder Is this backward compatible with the current version of `StanHeaders`/`rstan` on CRAN? If not, we can branch the code based on `USE_STANC3` macro.
```suggestion
 #ifdef USE_STANC3
 template <typename T2__, typename T5__>
 Eigen::Matrix<stan::promote_args_t<stan::value_type_t<T2__>,
 stan::value_type_t<T5__>>, -1, 1>
 csr_matrix_times_vector2(const int& m, const int& n, const T2__& w_arg__,
                          const std::vector<int>& v,
                          const std::vector<int>& u, const T5__& b_arg__,
                          std::ostream* pstream__) {
  return stan::math::csr_matrix_times_vector(m, n, w_arg__, v, u, b_arg__);
  #else
  template <typename T2__, typename T5__>
  inline
  Eigen::Matrix<typename boost::math::tools::promote_args<T2__, T5__>::type,
                Eigen::Dynamic, 1>
  csr_matrix_times_vector2(const int& m,
                           const int& n,
                           const Eigen::Matrix<T2__, Eigen::Dynamic, 1>& w,
                           const std::vector<int>& v,
                           const std::vector<int>& u,
                           const Eigen::Matrix<T5__, Eigen::Dynamic, 1>& b,
                           std::ostream* pstream__) {
    Eigen::Map<const Eigen::SparseMatrix<T2__,Eigen::RowMajor> >
      sm(m, n, w.size(), &u[0], &v[0], &w[0]);
    return sm * b;
  #endif
```

Reply:
Also, this is related to a specific version of `stanc3 > v2.26.1`, after https://github.com/stan-dev/stanc3/pull/865.

Reply:
Ah I did exactly this but idt we need the macro branch? I think the compiler should be able to figure out which version to use

Reply:
I think this is resolved now:
```r
* checking whether package ‘rstanarm’ can be installed ... OK
```
I'll re-check with the development version of `rstan`, after it's merged.